packages/server-wallet/src/wallet/index.ts

```diff
@@ -99,8 +100,21 @@ export class Wallet implements WalletInterface {
   async joinChannel(_args: JoinChannelParams): Result {
     throw 'Unimplemented';
   }
-  async updateChannel(_args: UpdateChannelParams): Result {
-    throw 'Unimplemented';
+  async updateChannel({channelId, appData, allocations}: UpdateChannelParams): Result {
+    return Channel.transaction(async tx => {
+      const channel = await Store.getChannel(channelId, undefined);
+
+      const outcome = deserializeAllocations(allocations);
+
+      const decision = updateChannel({channelId, appData, outcome}, channel);
+      if (Either.isLeft(decision)) {
+        throw decision.left;
```

Comment:
This is an example where we could make the code slightly more readable with naming:

```ts
const { left: err, right: newState } = ...;

if (err) {
    throw new Error(err);
}

Store.signState(newState);
```

Reply:
That would be nice, but I think the `Either.Either` type does not have _both_ a `left` and a `right` property: it has either the `left` property or the `right` property.

We can, of course, rename `Either.left` as `isError` and `Either.getLeft` somewhere, and write it like
```
if (isError(result) {
    throw getError(result);
}

Store.signState(getValue(result))
```

Here's another option:
```
      // This can be extracted as a helper
      const throwError = (error: Error) => {
        throw error;
      };
      const nextState = Either.getOrElseW(throwError)(
        updateChannel({channelId, appData, outcome}, channel)
      );
      const {outgoing, channelResult} = await Store.signState(nextState, tx);

      return {outbox: outgoing.map(n => n.notice), channelResults: [channelResult]};
```

Reply:
I tried the second strategy in 572a3ec.